src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/ImprisonmentReasonsResource.kt

```diff
@@ -0,0 +1,277 @@
+package uk.gov.justice.digital.hmpps.welcometoprison.resources
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.ArraySchema
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.welcometoprison.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrival.Arrival
+
+@RestController
+@Validated
+@RequestMapping(name = "Statuses", produces = [MediaType.APPLICATION_JSON_VALUE])
+class ImprisonmentReasonsResource {
+
+  @PreAuthorize("hasRole('ROLE_VIEW_ARRIVALS')")
+  @Operation(
+    summary = "Produces status and reason data",
+    description = "Produces imprisonment statuses and movement reasons data, requires ROLE_VIEW_ARRIVALS role",
+    security = [SecurityRequirement(name = "ROLE_VIEW_ARRIVALS", scopes = ["read"])],
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "JSON object of imprisonment statuses and movement reasons",
+        content = [
+          Content(
+            mediaType = "application/json",
+            array = ArraySchema(schema = Schema(implementation = Arrival::class))
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to retrieve",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "500",
+        description = "Unexpected error",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+    ]
+  )
+  @GetMapping(path = ["/imprisonment-statuses"], produces = [MediaType.APPLICATION_JSON_VALUE])
+  fun getStatuses(): List<ImprisonmentStatus> = imprisonmentStatuses
+}
+
+data class MovementReason(
+  val description: String? = "",
```

Comment:
Add `@JsonInclude(JsonInclude.Include.NON_NULL)` to the MovementReason and ImprisonmentStatus classes.
Also, add some swagger annotations to the fields to document their purpose / meaning.